lineapy/plugins/kubeflow_pipeline_writer.py

```diff
@@ -0,0 +1,166 @@
+    TaskDefinition,
+    TaskSerializer,
+    render_task_io_serialize_blocks,
+)
+from lineapy.plugins.taskgen import get_task_graph
+from lineapy.plugins.utils import load_plugin_template
+from lineapy.utils.logging_config import configure_logging
+from lineapy.utils.utils import prettify
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+
+class KubeflowDagFlavor(Enum):
+    ComponentPerSession = 1
+    ComponentPerArtifact = 2
+
+
+KubeflowDagConfig = TypedDict(
+    "KubeflowDagConfig",
+    {
+        "host_url": str,
+        "dag_flavor": str,  # Not native to DVC config
+    },
+    total=False,
+)
+
+
+class KubeflowPipelineWriter(BasePipelineWriter):
+    def _write_dag(self) -> None:
+        # Check if the given DAG flavor is a supported/valid one
+        try:
+            dag_flavor = KubeflowDagFlavor[
+                self.dag_config.get("dag_flavor", "ComponentPerArtifact")
+            ]
+        except KeyError:
+            raise ValueError(
+                f'"{dag_flavor}" is an invalid kubeflow dag flavor.'
+            )
+
+        # Construct DAG text for the given flavor
+        full_code = self._write_operators(dag_flavor)
+
+        # Write out file
+        file = self.output_dir / f"{self.pipeline_name}_dag.py"
+        file.write_text(full_code)
+        logger.info(f"Generated DAG file: {file}")
+
+    def _write_operators(
+        self,
+        dag_flavor: KubeflowDagFlavor,
+    ) -> str:
+        """
+        Returns a code block containing all the operators for a Kubeflow DAG.
+        """
+
+        DAG_TEMPLATE = load_plugin_template("kubeflow_dag.jinja")
+
+        if dag_flavor == KubeflowDagFlavor.ComponentPerSession:
+            task_breakdown = DagTaskBreakdown.TaskPerSession
+        elif dag_flavor == KubeflowDagFlavor.ComponentPerArtifact:
+            task_breakdown = DagTaskBreakdown.TaskPerArtifact
+
+        # Get task definitions based on dag_flavor
+        task_defs, _ = get_task_graph(
+            self.artifact_collection,
+            pipeline_name=self.pipeline_name,
+            task_breakdown=task_breakdown,
+        )
+
+        task_names = list(task_defs.keys())
+
+        task_defs = {tn: task_defs[tn] for tn in task_names}
+
+        (
+            rendered_task_defs,
+            task_loading_blocks,
+        ) = self.get_rendered_task_definitions(task_defs)
+
+        input_parameters_dict: Dict[str, Any] = {}
+        for parameter_name, input_spec in super().get_pipeline_args().items():
+            input_parameters_dict[parameter_name] = input_spec.value
+
+        full_code = DAG_TEMPLATE.render(
+            DAG_NAME=self.pipeline_name,
+            HOST_URL=self.dag_config.get("host_url", "http://localhost:3000"),
+            dag_params=input_parameters_dict,
+            task_definitions=rendered_task_defs,
+            tasks=task_defs,
+            task_loading_blocks=task_loading_blocks,
+            # task_dependencies=task_dependencies,
```

Comment:
clean up

Reply:
actually with the task teardown and cleanup I need this line now.